### PR TITLE
Fix Running TranslogOps on CS Thread

### PR DESCRIPTION
**Pull request by original-brownbear**
We should fork off from the CS thread to run this even if it's a rare condition.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java [hunk 1]
```diff
@@ -366,13 +367,11 @@ private void performTranslogOps(final RecoveryTranslogOperationsRequest request,
                 observer.waitForNextChange(new ClusterStateObserver.Listener() {
                     @Override
                     public void onNewClusterState(ClusterState state) {
-                        try {
+                        threadPool.generic().execute(ActionRunnable.wrap(listener, l -> {
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java:370]: Could also go for the write pool here (as suggested in https://github.com/elastic/elasticsearch/pull/58018) but I figured I wouldn't change that in this PR.

**Review by dnhatn — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java:370]: +1

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/default-distro (known issue)

**Comment by original-brownbear** [comment]
Thanks Nhat!

**Pull request merged**